lib/tasks/grda_warehouse.rake

```diff
@@ -12,6 +12,56 @@ namespace :grda_warehouse do
   desc 'Setup a sample GRDA warehouse database'
   task setup: [:migrate, :seed_data_sources]
 
+  namespace :ssn do
+    desc 'Report SSN changes that would result from the enhanced selector'
+    task :audit, [:use_oldest] => [:environment, 'log:info_to_stdout'] do |_task, args|
+      require 'progress_bar'
+      use_oldest_arg = args[:use_oldest]
+      use_oldest = use_oldest_arg.nil? || ActiveModel::Type::Boolean.new.cast(use_oldest_arg)
+
+      total = GrdaWarehouse::Hud::Client.destination.count
+      progress = ProgressBar.new(total, :counter, :bar, :percentage, :rate, :eta)
+
+      csv_output = CSV.generate(headers: true) do |csv|
+        csv << ['client_id', 'current_ssn', 'current_ssn_dq', 'proposed_ssn', 'proposed_ssn_dq']
+
+        default_dq = 99
+        default_date_created = 10.years.ago
+        data_source_ids = GrdaWarehouse::DataSource.pluck(:id)
+
+        GrdaWarehouse::Hud::Client.destination.preload(:source_clients).find_in_batches(batch_size: 1000) do |clients|
+          clients.each do |destination|
+            progress.increment!
+            source_clients = destination.source_clients.filter_map do |source|
+              next unless source.data_source_id.in?(data_source_ids)
+
+              source.SSNDataQuality ||= default_dq
+              source.DateCreated ||= default_date_created
+              source
+            end
+
+            next if source_clients.empty?
+
+            dest_attr = destination.attributes.with_indifferent_access.slice(:SSN, :SSNDataQuality)
+            proposed = GrdaWarehouse::SSNSelector.call(dest_attr: dest_attr.dup, source_clients: source_clients, use_oldest: use_oldest)
+
+            next if [destination.SSN, destination.SSNDataQuality] == [proposed[:SSN], proposed[:SSNDataQuality]]
```

Comment:
So this will also report on cases where SSN is unchanged but SSNDataQuality was "downgraded"?  Maybe that is intended, just wanted to flag that I think it will lead to more data to sift through to find cases where SSN actually changed. Again maybe thats fine, not sure how we are intending to use the output of this. 

For example-- if I'm understanding correctly-- if there is a Destination client with 1 source client with values `SSN: "XXXXX1234", SSNDataQuality: 1`, it will be flagged by this process because the SSNSelector would return `2` as the data quality value.

Reply:
Yes, that's accurate. I guess if the dq changes it seems worth having visibility into that. Assuming downstream consumers care about dq but maybe it's generally ignored.

I'd probably do some quick analysis on the output to count records where dq changed, count where ssn changed. That should give us and idea of the impact.

Reply:
Also note dq can be upgraded to 2 if the record has a numeric ssn and dq was 99 or something